src/FTPClientWrapperSSH.cpp

```diff
@@ -655,6 +622,8 @@ int FTPClientWrapperSSH::verify_knownhost(ssh_session session) {
 	if (rc < 0)
 		return -1;
 
+	const char *keytype = ssh_key_type_to_char(ssh_key_type(srv_pubkey));

```

Comment:
Sure, but it's not possible to get an unknown type host key. The client picks the best (supported) key offered by the server. Unknown keys aren't going to be selected. If the server does not present a key that the client can handle we won't get this far.